lockable/provider_http.py

```diff
@@ -0,0 +1,41 @@
+""" resources Provider for HTTP """
+import logging
+import requests
+from requests import HTTPError
+
+from lockable.provider import Provider, ProviderError
+
+MODULE_LOGGER = logging.getLogger('lockable')
+
+
+class ProviderHttp(Provider):
+    """ ProviderHttp interface"""
+
+    def __init__(self, uri: str):
+        """ ProviderHttp constructor """
+        super().__init__(uri)
+        self._reload()
+
+    def _reload(self) -> None:
+        """ Reload resources list from web server """
+        self.set_resources_list(self._get_http(self._uri))
+
+    @staticmethod
+    def _get_http(uri: str) -> list:
+        """ Internal method to get http json data"""
+        try:
+            response = requests.get(uri)
+            response.raise_for_status()
+
+            # could utilise ETag or Last-Modified headers to optimize performance
+            # etag = response.headers.get("ETag")
+            # last_modified = response.headers.get("Last-Modified")
+
+            # access JSON content
+            return response.json()
+        except HTTPError as http_err:
```

Comment:
This probably results to whole test job failing? Should this implement internal retry logic?

Reply:
I was thinking the same. Perhaps it would be good idea to retry at least some cases.

Reply:
retry logic perhaps could be introduce follow-up PR since its ”feature” and not mandatory here.

Reply:
coming here: https://github.com/jupe/py-lockable/pull/16